### PR TITLE
Error Handling Features

### PR DESCRIPTION
**Pull request by zachdt**
### Description

Opening this PR before making changes to the order card.

- Added custom fallback to ErrorBoundary 
- Added `useError` hook for async error catching in DOM
- Added Tooltip component for displaying messages on mouse-over

### CODE REVIEW

**Comment by zachdt** [comment]
Solving a TS bug on the Error Boundary, not ready to go quite yet.

File: app/src/components/Market/OptionsTable/OptionsTable.tsx [hunk 1]
```diff
@@ -93,13 +95,13 @@ const OptionsTable: React.FC<OptionsTableProps> = (props) => {
...
-    'Reserve',
-    'Contract',
-    '',
+    { name: 'Strike Price', tip: null },
+    { name: 'Break Even', tip: null },
+    { name: 'Price', tip: null },
+    { name: '2% Depth', tip: '# of options can be bought at <2% slippage' },
+    { name: 'Reserve', tip: null },
+    { name: 'Contract', tip: null },
+    { name: '', tip: null },
```
Comment (Alexangelj) [inline app/src/components/Market/OptionsTable/OptionsTable.tsx:104]: Nice

File: app/src/hooks/utils/useError.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export const useError = () => {
+  const [_, setError] = useState()
+
+  return useCallback(
+    (e) => {
+      setError(() => {
+        throw e
+      })
+    },
+    [setError]
```
Comment (Alexangelj) [inline app/src/hooks/utils/useError.ts:12]: Needed this

**Review by Alexangelj — APPROVED** [review]
LGTM

**Pull request merged**